### ISSUE TITLE
Feature idea: background builds of dependencies

### ISSUE THREAD

**Issue by smmalis37** [rust-analyzer/rust-analyzer#5834]
The idea here is an off-by-default setting that would, once rust-analyzer is loaded and has finished performing the initial `cargo check`, perform a `cargo build` of all dependencies while the user is writing code. In this way, once they decide their code is ready to run, they don't have to wait for dependencies to compile, they'd already be done. Configuration could include which profiles to build ahead of time (dev, release, test, bench, custom). It would also be nice if this process could be paused on save to ensure the `cargo check`s are providing up-to-date diagnostics, but I don't know how feasible that is (or if it's even necessary).

**Comment by matklad** [comment]
I think this is out of scope for ra, as we don't generally manage build system.
